pkg/web/router/router.go

```diff
@@ -28,5 +26,102 @@ func NewRouter(config config.Config) *echo.Echo {
 }
 
 func SetAppRoutes(r *echo.Group, appsHandler *apps.HTTPHandler) {
+	// swagger:operation GET /apps App
+	//
+	// Returns root level information for all apps
+	// ---
+	// summary: Retrieve list of apps
+	// operationId: getApps
+	// produces:
+	// - application/json
+	// parameters: []
+	// responses:
+	//   200:
+	//     description: successful operation
+	//     schema:
+	//       $ref: '#/definitions/App'
 	r.GET("/apps", appsHandler.GetApps)
+
+	// swagger:operation GET /apps/{id} App
+	//
+	// Retrieve all information for a single app including all child information
+	// ---
+	// summary: Get app by id
+	// operationId: getAppById
+	// produces:
+	// - application/json
+	// parameters:
+	// - name: id
+	//   in: path
+	//   description: The id for the app that needs to be fetched.
+	//   required: true
+	//   type: string
+	// responses:
+	//   200:
+	//     description: successful operation
+	//     schema:
+	//       $ref: '#/definitions/App'
+	//   400:
+	//     description: Invalid id supplied
+	//   404:
+	//     description: App not found
+	r.GET("/apps/{id}", appsHandler.GetAppById) // TODO: Implement correctly the call of the method passing the parameters
```

Comment:
@camilamacedo86 IS it possible for go-swagger to take more from the code. I always see the major benefit or auto-generated API docs, that we don't need to manually update information which is an additional step. With the comments here, we will need to update it if additional responses are added or other modifications.

Reply:
@davidffrench in other languages as JAVA we can find the injections annotations to do it. The swagger will not parse the router implementation. It will parse comments and/or notations in all languages. 

However, in the case of the model, it is able to parse the struct as you can see [here](https://github.com/aerogear/mobile-security-service/pull/40/files#diff-d2e6ed7f6795bd3fdbcb783f1c3850a5R4). 

I liked the [beego](https://beego.me/) framework and how it works with swagger. It is using annotations which are more similar to java see [here](https://beego.me/blog/beego_api) how to it is done. However, for this project was adopted [echo](https://echo.labstack.com/) framework which is minimalist and has not this kind of feature. 

In this way, shows that the best approach is to use the [go-swagger](https://github.com/go-swagger/go-swagger) which is the same way that is done in the [aerogear-app-metrics](https://github.com/aerogear/aerogear-app-metrics/blob/master/pkg/web/router.go). 

Please, let me know if it still missing some clarification and/or you have any objection here. Also, please let me know if we can move forward and merge this pr.

Reply:
Thanks for the clarification @camilamacedo86 . I am weighing up the pros and cons of including this.

*Pros*
- Pick up changes to a model on regeneration
- Swagger spec is easier to edit as it is now commented in our code

*Cons*
- Some models have fields which will never be returned in the response. This is confusing for someone just looking at the swagger api.
- The comments are verbose and still need to be updated manually for additional changes

Reply:
Hi @davidffrench,

Thank you for your inputs.  Following some considerations regards the cons added here.

Regards 

> Some models have fields which will never be returned in the response. This is confusing for someone just looking at the swagger api

* The struct/objects mapped in the Swagger are just that ones which will be used in the endpoints and not all of them. 
* The metatag `swagger:model` should be added just in the structs that will be used by the API. 
* All struct that will be used as input/output of the RestAPI should be mapped and these structs don't necessary will be the database model. For example, an `/search` endpoint can return a list of items which will be exactly the database model or a list of items with a type that is constructed just for the response with just the relevant properties. (e.g ItemTO)

Regards

> The comments are verbose and still need to be updated manually for additional changes
* Besides I agree with the fact that it is verbose, it still lower than defined the YAML file manually as it was done before
* For routers the verbose are kind of similar, however, over the models, it is really better

Regards the pros added here.

> _Pros_
> 
> * Pick up changes to a model on regeneration
> * Swagger spec is easier to edit as it is now commented in our code   

I agree with them and I would like to add.

* It makes more intuitive the need of the update in the swagger syntax when a change is made
* It is verbose but less than defined the YAML file manually as it was
* The syntax is a common partner used and is at least intuitive besides for the router still verbose. 
* It is the commonly used approach was I could see in the projects with this stack/frameworks adopted
* It is done in the same way that we can find in the aerogear projects.

Reply:
Hi @davidffrench,

Thank you for your inputs.  Following some considerations and ideas.

**Regards _CONS_**

1. > Some models have fields which will never be returned in the response. This is confusing for someone just looking at the swagger api

* The struct/objects mapped for the Swagger API are just that ones which will be used in the endpoints and not all of them. 
* The metatag `swagger:model` should be added just in the structs that will be used by the API. 
* All struct that will be used as input/output of the RestAPI should be mapped and these structs don't necessary will be the database model. For example, an `/search` endpoint can return a list of items which will be exactly the database model struct or a list of items with a type that is constructed just for the input/response with the relevant properties only. (e.g ItemTO)

2. > The comments are verbose and still need to be updated manually for additional changes
* Besides I agree with the fact that it still verbose for the routers. It is less verbose and hard than define the full YAML file manually as it was done before

**Regards _PROS_**

> * Pick up changes to a model on regeneration
> * Swagger spec is easier to edit as it is now commented in our code   

I agree with them and I would like to add.

* It makes more intuitive the need of update the swagger syntax when a change is made
* It is verbose but less than defined the YAML file manually as it was
* The syntax is a commonly used and is at least intuitive besides for the router still verbose. 
* It is the commonly used approach that I could see in the projects with the same stack/frameworks adopted
* It is done in the same way that we can find in the aerogear projects.

So, wdyt? Could we move forward here or you would like to keep it manually done as it is?

Reply:
Thank you and I agree, lets move forward with this.